internal/blocksync/pool.go

```diff
@@ -233,10 +234,11 @@ func (pool *BlockPool) PopRequest() {
 		delete(pool.requesters, pool.height)
 		pool.height++
 		pool.lastAdvance = time.Now()
-
 		// the lastSyncRate will be updated every 100 blocks, it uses the adaptive filter
 		// to smooth the block sync rate and the unit represents the number of blocks per second.
-		if (pool.height-pool.startHeight)%100 == 0 {
+		// -1 because the start height is assumed to be 1  @jmalicevic ToDo, verify it is still OK when
+		// starting height is not 1
+		if (pool.height-pool.startHeight-1)%100 == 0 && pool.height != pool.startHeight {
```

Comment:
I may be missing the point here but, like it is now (and for a `startHeight` of 1), won't the condition become true for height 2?

Reply:
I f we want to really count every 100 blocks and we include the block at `initialHeight` (which is also synced), the correct condition is in fact `(pool.height - pool.startHeight + 1) % 100`. Additionally, for tests not to time out, we need to set the height of the chain ( `pool.maxPeerHeigth`) to be 100 + the maximum initial height of the testnet nodes. Otherwise the sync rate is not calculated properly.